sklearn/ensemble/_iforest.py

```diff
@@ -477,19 +489,20 @@ def _compute_score_samples(self, X, subsample_features):
 
         depths = np.zeros(n_samples, order="f")
 
-        for tree, features in zip(self.estimators_, self.estimators_features_):
+        for tree_idx, (tree, features) in enumerate(
+            zip(self.estimators_, self.estimators_features_)
+        ):
             X_subset = X[:, features] if subsample_features else X
 
             leaves_index = tree.apply(X_subset)
```

Comment:
It should be safe to call `apply` with `check_input=False` because we already validated in the public function.

Reply:
Still got `ValueError: X.dtype should be np.float32, got float64` when tried with check_input=False.

Reply:
Interesting. I have to further check the reason for that.

Reply:
It was a missing `dtype=tree_dtype` parameter in the call to `_validate_data` at prediction type. It was there only at `fit` time. I pushed a commit.

Reply:
Thanks for having taken a look at it. Nice that it was not a deeper issue than that.